### PR TITLE
Navy 207

### PR DESCRIPTION
**Pull request by melgenek**
Short description if any.

Author Todo List:

- [x] Add/adjust tests (if applicable)
- [x] Build in CI passes
- [x] Latest master revision is merged into the branch
- [x] Self-Review
- [x] Set `Ready For Review` status

### CODE REVIEW

File: src/idex/identity-resolver.js [hunk 1]
```diff
@@ -2,10 +2,20 @@ import { toParams } from '../utils/url'
 import { fromError } from '../utils/emitter'
 import { expiresInHours, asParamOrEmpty, asStringParam, mapAsParams } from '../utils/types'
 import { DEFAULT_IDEX_EXPIRATION_HOURS, DEFAULT_IDEX_AJAX_TIMEOUT, DEFAULT_IDEX_URL } from '../utils/consts'
+import { base64UrlEncode } from '../utils/b64'
 
 const IDEX_STORAGE_KEY = '__li_idex_cache'
 
-function _responseReceived (storageHandler, domain, expirationHours, successCallback) {
+function _cacheKey (additionalParams) {
```
Comment (melgenek) [inline src/idex/identity-resolver.js:9]: this is the only thing that is needed to make the cache work for different params.
Everything else is fixing the linting and making the storage handler generic.

File: src/handlers/storage-handler.js [hunk 2]
```diff
@@ -47,5 +54,39 @@ export function StorageHandler (storageStrategy, externalStorageHandler) {
...
+
+  const genericFunctions = {
+    get: key => {
+      if (strEqualsIgnoreCase(storageStrategy, StorageStrategy.none)) {
+        return null
+      } else if (strEqualsIgnoreCase(storageStrategy, StorageStrategy.localStorage)) {
+        if (functions.localStorageIsEnabled()) {
+          const expirationKey = `${key}_exp`
+          const oldLsExpirationEntry = functions.getDataFromLocalStorage(expirationKey)
+          if (oldLsExpirationEntry && Date.parse(oldLsExpirationEntry) <= new Date().getTime()) {
```
Comment (3link) [inline src/handlers/storage-handler.js:66]: This will always expire entries.

**Review by 3link — CHANGES_REQUESTED** [review]

File: src/handlers/storage-handler.js (hunk 2 above)
Comment (3link) [reply src/handlers/storage-handler.js:66]: Never mind, misread it.

**Review by 3link — CHANGES_REQUESTED** [review]

File: src/idex/identity-resolver.js [hunk 3]
```diff
@@ -17,11 +27,10 @@ function _responseReceived (storageHandler, domain, expirationHours, successCall
...
-      storageHandler.setCookie(
-        IDEX_STORAGE_KEY,
+      storageHandler.set(
+        _cacheKey(additionalParams),
         JSON.stringify(responseObj),
         expiresInHours(expirationHours),
-        'Lax',
         domain)
     } catch (ex) {
       console.error('Error storing response to cookies', ex)
```
[the remark this answers is not in the document]
Comment (melgenek) [reply src/idex/identity-resolver.js]: https://github.com/LiveIntent/live-connect/pull/53/commits/7a72beb052d4332a7ea383953200d44feb8ce20f#diff-a57fbdfbf75933739497179697efad2140acc7c7d05c6cd3249b76646e5b7592L36

File: src/idex/identity-resolver.js [hunk 4]
```diff
@@ -63,11 +72,11 @@ export function IdentityResolver (config, storageHandler, calls) {
       return `${url}/${source}/${publisherId}${params}`
     }
     const unsafeResolve = (successCallback, errorCallback, additionalParams) => {
-      const storedCookie = storageHandler.getCookie(IDEX_STORAGE_KEY)
+      const storedCookie = storageHandler.get(_cacheKey(additionalParams))
```
[the remark this answers is not in the document]
Comment (melgenek) [reply src/idex/identity-resolver.js]: https://github.com/LiveIntent/live-connect/pull/53/commits/7a72beb052d4332a7ea383953200d44feb8ce20f#diff-a57fbdfbf75933739497179697efad2140acc7c7d05c6cd3249b76646e5b7592R74

File: src/manager/identifiers.js [hunk 5]
```diff
@@ -35,52 +34,19 @@ export function resolve (state, storageHandler) {
...
-        emitter.error('CookieGetOrAdd', 'Failed manipulating cookie jar', e)
-      }
-      return ret
-    }
-
-    const getOrAddWithExpiration = (key, value, storageOptions, storageStrategy) => {
-      if (strEqualsIgnoreCase(storageStrategy, StorageStrategy.localStorage)) {
-        return lsGetOrAdd(key, value, storageOptions)
-      } else if (strEqualsIgnoreCase(storageStrategy, StorageStrategy.none)) {
+        emitter.error('CookieLsGetOrAdd', 'Failed manipulating cookie jar or ls', e)
```
[the remark this answers is not in the document]
Comment (3link) [reply src/manager/identifiers.js:48]: I see.

**Review by 3link — APPROVED** [review]

**Pull request merged**